tests/e2e/specs/miscellaneous/VsixInstallationDisableTest.spec.ts

```diff
@@ -0,0 +1,283 @@
+		const contextMenuAvailable: boolean = await explorerView.isContextMenuItemVisible('Install Extension VSIX');
+		await explorerView.closeContextMenu();
+
+		expect(contextMenuAvailable).to.equal(
+			expectedVisible,
+			`Explorer context menu should ${expectedVisible ? 'contain' : 'not contain'} Install Extension VSIX action`
+		);
+	}
+
+	/**
+	 * runs all UI checks for VSIX installation capability
+	 * Verifies command palette, extensions menu, and explorer context menu
+	 * @param expectedEnabled - Whether VSIX installation should be enabled or not
+	 */
+	async function checkVsixInstallationInUI(expectedEnabled: boolean): Promise<void> {
+		Logger.info(`Checking VSIX installation capability in UI - expected enabled: ${expectedEnabled}`);
+
+		await checkCommandPaletteForVsixCommand(expectedEnabled);
+		await checkExtensionsMenuForVsixAction(expectedEnabled);
+		await checkExplorerContextMenuForVsixAction(expectedEnabled);
+	}
+
+	/**
+	 * verifies whether default VSIX extensions were auto-installed or not
+	 * @param shouldBeInstalled - Whether the extensions should be installed or not
+	 */
+	async function checkDefaultExtensionsInstallation(shouldBeInstalled: boolean): Promise<void> {
+		Logger.info(`Checking default VSIX extensions auto-installation - expected installed: ${shouldBeInstalled}`);
+
+		await extensionsView.openExtensionsView();
+		const installedExtensions: string[] = await extensionsView.getInstalledExtensionNames();
+
+		Logger.debug(`Found installed extensions: ${installedExtensions.join(', ')}`);
+
+		for (const extensionName of DEFAULT_EXTENSIONS) {
+			const isInstalled: boolean = installedExtensions.some((installed: string): boolean =>
+				installed.toLowerCase().includes(extensionName.toLowerCase())
+			);
+
+			expect(isInstalled).to.equal(
+				shouldBeInstalled,
+				`Default VSIX extension "${extensionName}" should ${shouldBeInstalled ? 'be auto-installed' : 'not be auto-installed'}`
+			);
+		}
+	}
+
+	/**
+	 * applies a ConfigMap from a file to the cluster
+	 * @param configFileName - Name of the ConfigMap file to apply
+	 */
+	function applyConfigMap(configFileName: string): void {
+		const configPath: string = path.join(RESOURCES_PATH, configFileName);
+		const configContent: string = fs.readFileSync(configPath, 'utf8');
+		shellExecutor.executeCommand(`oc apply -f - <<EOF\n${configContent}\nEOF`);
+	}
+
+	/**
+	 * deletes a ConfigMap from the cluster
+	 * @param name - Name of the ConfigMap
+	 * @param namespace - Namespace of the ConfigMap
+	 */
+	function deleteConfigMap(name: string, namespace: string): void {
+		Logger.debug(`Deleting ConfigMap ${name} from namespace ${namespace}`);
+		shellExecutor.executeCommand(`oc delete configmap ${name} -n ${namespace} --ignore-not-found=true`);
+	}
+
+	/**
+	 * removes all ConfigMaps created by the test
+	 */
+	function cleanupConfigMaps(): void {
+		deleteConfigMap(CONFIG_MAP_NAMES.VSIX_CONFIG, CONFIG_MAP_NAMESPACES.VSIX_CONFIG);
```

Comment:
IMHO, extracting code into deleteConfigMap() makes it harder to read and understand.